apps/api/tests/api/application-state.test.ts

```diff
@@ -191,24 +228,34 @@ describe('State Machine', () => {
 		});
 
 		it('should change from DAC_REVIEW to REJECTED on rejected', async () => {
-			const result = await createApplicationStateManager({ id: 2 });
+			const result = await createApplicationStateManager({ id: 4 });
 			assert.ok(result.success);
 
 			const dacReviewManager = result.data;
 			await dacReviewManager.submitDraft();
 			await dacReviewManager.approveRepReview();
-
 			stateValue = dacReviewManager.getState();
 			assert.strictEqual(stateValue, DAC_REVIEW);
 
 			await dacReviewManager.rejectDacReview();
-
 			stateValue = dacReviewManager.getState();
 			assert.strictEqual(stateValue, REJECTED);
+
+			const actionResult = await testActionRepo.listActions({ application_id: 4 });
+			assert.ok(actionResult.success && actionResult.data);
+
+			assert.ok(
+				actionResult.data.find(
+					(record) =>
+						record.action === ApplicationActions.DAC_REVIEW_REJECTED &&
+						record.state_before === ApplicationStates.DAC_REVIEW &&
+						record.state_after === ApplicationStates.REJECTED,
+				),
+			);
```

Comment:
Additional check for DB Action records added to a handful of state transition tests